subsys/net/ip/net_if.c

```diff
@@ -527,16 +535,49 @@ static void leave_mcast_all(struct net_if *iface)
 #define leave_mcast_all(...)
 #endif /* CONFIG_NET_IPV6_MLD */
 
+static bool timer_check_timeout(u64_t start, u32_t time, u64_t current_time)
+{
+	if ((s64_t)((start + time) - current_time) > 0) {
+		return false;
+	}
+
+	return true;
+}
+
+static bool timer_timedout(u64_t timer_start, u32_t timer_timeout,
+			   u64_t current_time)
+{
+	return timer_check_timeout(timer_start, timer_timeout, current_time);
+}
+
+typedef void (*timer_expired_cb_t)(struct net_if_addr *ifaddr);
+
+static u32_t timer_manage_timeout(struct net_if_addr *ifaddr,
+				  u64_t timer_start,
+				  u32_t timer_timeout,
+				  u64_t current_time,
+				  timer_expired_cb_t timer_expired_cb)
+{
+	s32_t next_timeout;
+
+	if (timer_timedout(timer_start, timer_timeout, current_time)) {
+		timer_expired_cb(ifaddr);
+		return UINT32_MAX;
+	}
+
+	next_timeout = current_time - (timer_start + timer_timeout);
+	return abs(next_timeout);
```

Comment:
As argued yesterday, no need to use abs() when working with events ordered in time in known way. And as there's no need for them, then if you have them, that's highly confusing. Sorry, I'm firm on this, and that's the reason for -1.

Here, you tested timer is not timed out. So, you know that current_time is less than (timer_start + timer_timeout). So, just use unsigned arithmetics and expression of `(timer_start + timer_timeout) - current_time`.